js/src/components/PrimaryTaxonomyPicker.js

```diff
@@ -0,0 +1,161 @@
+
+const PrimaryTaxonomyPickerLabel = styled.label`
+	padding-top: 16px;
+`;
+
+/**
+ * A component for selecting a primary taxonomy term.
+ */
+class PrimaryTaxonomyPicker extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.onChange = this.onChange.bind( this );
+		this.updateReplacementVariable = this.updateReplacementVariable.bind( this );
+
+		const { field_id: fieldId, name } = props.taxonomy;
+		this.input = document.getElementById( fieldId );
+		props.setPrimaryTaxonomy( name, parseInt( this.input.value, 10 ) );
+	}
+
+	/**
+	 * Updates the replacement variable when the terms were not yet retrieved on mount.
+	 *
+	 * @param {Object} prevProps The previous props.
+	 *
+	 * @returns {void}
+	 */
+	componentDidUpdate( prevProps ) {
+		// Update replacement variable when taxonomy has not yet been retrieved on mount.
+		if ( prevProps.terms.length === 0 && this.props.terms.length > 0 ) {
+			this.updateReplacementVariable( this.props.primaryTaxonomy );
+		}
+	}
+
+	/**
+	 * Handles an onChange event.
+	 *
+	 * Updates the primary taxonomy in the store, as well as the replacement variable and the hidden field.
+	 *
+	 * @param {number} termId The term's id.
+	 *
+	 * @returns {void}
+	 */
+	onChange( termId ) {
+		const { name } = this.props.taxonomy;
+
+		this.updateReplacementVariable( termId );
+
+		this.props.setPrimaryTaxonomy( name, termId );
+
+		this.input.value = termId;
+	}
+
+	/**
+	 * Updates the primary taxonomy replacement variable.
+	 *
+	 * @param {number} termId The term's id.
+	 *
+	 * @returns {void}
+	 */
+	updateReplacementVariable( termId ) {
+		if ( this.props.taxonomy.name !== "category" ) {
+			return;
+		}
+		const primaryTerm = this.props.terms.find( term => term.id === termId );
+		if ( primaryTerm ) {
+			this.props.updateReplacementVariable( `primary_${ this.props.taxonomy.name }`, primaryTerm.name );
+		}
+	}
+
+	/**
+	 * Renders the PrimaryTaxonomyPicker component.
+	 *
+	 * @returns {ReactElement} The rendered component.
+	 */
+	render() {
+		const {
+			primaryTaxonomy,
+		} = this.props;
+
+		return (
+			<div className="components-base-control__field">
+				<PrimaryTaxonomyPickerLabel
+					htmlFor="yoast-primary-category-picker"
+					className="components-base-control__label">
+					{
+						sprintf(
+							/* Translators: %s: category name */
+							__( "Select the primary %s" ),
+							this.props.taxonomy.singular_label.toLowerCase()
+						)
+					}
+				</PrimaryTaxonomyPickerLabel>
+				<TaxonomyPicker
+					value={ primaryTaxonomy }
+					onChange={ this.onChange }
+					id="yoast-primary-category-picker"
+					terms={ this.props.terms }/>
+			</div>
+		);
+	}
+}
+
+PrimaryTaxonomyPicker.propTypes = {
+	terms: PropTypes.array,
+	primaryTaxonomy: PropTypes.string,
+	setPrimaryTaxonomy: PropTypes.func,
+	updateReplacementVariable: PropTypes.func,
+	taxonomy: PropTypes.shape( {
+		name: PropTypes.string,
+		/* eslint-disable */
```

Comment:
Please disable only the specific rule that is broken. There is no need to disable eslint entirely.

Reply:
The ESLint rule we use is bugged, using `/* eslint-disable camelcase */` gives an eslint warning ironically.